qiskit_experiments/library/tomography/fitters/cvxpy_lstsq.py

```diff
@@ -181,45 +153,17 @@ def cvxpy_linear_lstsq(
     #                   + 1j * (bm_r * vec(rho_i) + bm_i * vec(rho_r))
     #                 = bm_r * vec(rho_r) - bm_i * vec(rho_i)
     # where we drop the imaginary part since the expectation value is real
-
     bm_r = np.real(basis_matrix)
     bm_i = np.imag(basis_matrix)
-
-    # SDP objective function
     arg = bm_r @ cvxpy.vec(rho_r) - bm_i @ cvxpy.vec(rho_i) - probability_data
     obj = cvxpy.Minimize(cvxpy.norm(arg, p=2))
+    prob = cvxpy.Problem(obj, cons)
 
     # Solve SDP
-    prob = cvxpy.Problem(obj, cons)
-    iters = 5000
-    max_iters = kwargs.get("max_iters", 20000)
-    # Set default solver if none is specified
-    if "solver" not in kwargs:
-        if "CVXOPT" in cvxpy.installed_solvers():
-            kwargs["solver"] = "CVXOPT"
-        elif "MOSEK" in cvxpy.installed_solvers():
-            kwargs["solver"] = "MOSEK"
-
-    problem_solved = False
-    while not problem_solved:
-        kwargs["max_iters"] = iters
-        prob.solve(**kwargs)
-        if prob.status in ["optimal_inaccurate", "optimal"]:
-            problem_solved = True
-        elif prob.status == "unbounded_inaccurate":
-            if iters < max_iters:
-                iters *= 2
-            else:
-                raise AnalysisError(
-                    "CVXPY fit failed, probably not enough iterations for the " "solver"
-                )
-        elif prob.status in ["infeasible", "unbounded"]:
-            raise AnalysisError(
-                "CVXPY fit failed, problem status {} which should not " "happen".format(prob.status)
-            )
-        else:
-            raise AnalysisError("CVXPY fit failed, reason unknown")
+    cvxpy_utils.set_default_sdp_solver(kwargs)
+    cvxpy_utils.solve_iteratively(prob, 5000, **kwargs)
```

Comment:
Not really unless we make a separate analysis class for each fitter function since its a specific detail to the cvxpy fit function.

The 5000 was used as an initial value, and if it isn't solved by then it repeats with a larger number up to the cvxpy solver option value for `max_iters` (these initial parameter values were chosen heuristically at some point during ignis development)